iceoryx_posh/test/moduletests/test_mepoo_chunk_header.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/mepoo/chunk_header.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::mepoo;
+
+template <uint32_t N>
+struct alignas(32) ChunkWithPayload {
+    ChunkHeader m_chunkHeader;
+    uint8_t m_payload[N];
+};
+
+template <uint32_t N>
+struct alignas(32) ChunkWithCustomHeaderAndPayload {
+    ChunkHeader m_chunkHeader;
+    uint64_t m_customHeader;
+    uint8_t m_payload[N];
+};
+
+class ChunkHeader_test : public Test
+{
+    void SetUp(){};
+    void TearDown(){};
+};
+
+TEST_F(ChunkHeader_test, ChunkHeaderHasInitializedMembers)
+{
+    ChunkHeader sut;
+
+    EXPECT_THAT(sut.m_chunkSize, Eq(0U));
+
+    // deliberately used a magic number to make the test fail when CHUNK_HEADER_VERSION changes
+    EXPECT_THAT(sut.m_chunkHeaderVersion, Eq(1U));
+
+    EXPECT_THAT(sut.m_reserved1, Eq(0U));
+    EXPECT_THAT(sut.m_reserved2, Eq(0U));
+    EXPECT_THAT(sut.m_reserved3, Eq(0U));
+
+    EXPECT_THAT(sut.m_originId, Eq(iox::UniquePortId(iox::popo::CreateInvalidId)));
+
+    EXPECT_THAT(sut.m_sequenceNumber, Eq(0U));
+
+    EXPECT_THAT(sut.m_payloadSize, Eq(0U));
+    // a default created ChunkHeader has always an adjacent payload
+    EXPECT_THAT(sut.m_payloadOffset, Eq(sizeof(ChunkHeader)));
+}
+
+TEST_F(ChunkHeader_test, CustomHeaderMethodReturnsCorrectCustomHeaderPointer)
+{
+    constexpr int32_t PAYLOAD_SIZE {128};
```

Comment:
What is with the cases:
1. Payload size is 0. Do we allow or forbid this?
2. Payload size is UINT32_MAX?
3. A test where m_payloadSize + m_chunkHeaderSize > UINT32_MAX

Reply:
not relevant here, I created a dummy chunk and checked if the method works as expected. All the stuff you mentioned is not set by the `ChunkHeader` but by the ports. This might change with the removal of `ChunkInfo`.
Currently there is no way to create a custom header. I just added the method to have the `ChunkHeader` API ready. If it's a problem, I can remove it and add it again when the actual functionality is implemented.

Reply:
This tests will be added when a method is added where the payload size is used, e.g. `getUsedSizeOfChunk`. Currently there is a test that checks that the payload size is large enough for a mempool chunk